### PR TITLE
Update gemfile

### PR DESCRIPTION
**Pull request by Steveo1485**

### CODE REVIEW

**Comment by Steveo1485** [comment]
Accidentally did pull request on gopher-snakes-2013 repo and not my fork. Please ignore.

**Comment by zspencer** [comment]
Fine to do the PR against this repo; but don't merge it ;)

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+get '/' do
+  @all_postings = Post.all
+  erb :index
+end
+
+post '/' do
+  edit_url = ""
+  10.times do
+    edit_url << (rand(9) + 1).to_s
+  end
```
Comment (zspencer) [inline app.rb:24]: Look into the SecureRandom module to create random strings: http://www.ruby-doc.org/stdlib-1.9.3/libdoc/securerandom/rdoc/SecureRandom.html

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  new_post = Post.create(title: params["title"], description: params["description"], price: params["price"], edit_url: edit_url)
+  redirect("/#{new_post.id}/#{new_post.edit_url}")
+end
+
+get '/:id' do
+  @post_id = params[:id]
+  @posting = Post.find_by(id: @post_id)
+  erb :posting
```
Comment (zspencer) [inline app.rb:32]: Not sure why the `@post_id variable` needs to exist. Couldn't you just access the `@posting.id` anywhere you need it?

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  erb :index
+end
+
+post '/' do
+  edit_url = ""
+  10.times do
+    edit_url << (rand(9) + 1).to_s
+  end
+  new_post = Post.create(title: params["title"], description: params["description"], price: params["price"], edit_url: edit_url)
+  redirect("/#{new_post.id}/#{new_post.edit_url}")
```
Comment (zspencer) [inline app.rb:26]: What happens if the post creation fails validations?

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+get '/:id' do
+  @post_id = params[:id]
+  @posting = Post.find_by(id: @post_id)
+  erb :posting
+end
+
+get '/:id/:edit_url' do
+  @post_id = params[:id]
+  @posting = Post.find_by(id: @post_id)
+  erb :edit_posting
```
Comment (zspencer) [inline app.rb:38]: ee previous comment on not sure why `@post_id` needs to exist. Seems to violate rule 4 of fewer moving parts.

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  redirect("/#{new_post.id}/#{new_post.edit_url}")
+end
+
+get '/:id' do
+  @post_id = params[:id]
+  @posting = Post.find_by(id: @post_id)
+  erb :posting
+end
+
+get '/:id/:edit_url' do
```
Comment (zspencer) [inline app.rb:35]: I would recommend keeping your posts behind a `/posts` route, as routes which start with a matcher can cause issues. Specifically, they greedily match.

File: spec/features/edit_listing_spec.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+feature "Edit a listing" do
+  before :each do
+    visit('/')
+    fill_in('title', with: 'iPad')
+    fill_in('description', with: 'White 16GB iPad 2. Wifi with Orange cover')
+    fill_in('price', with: '200')
+    click_button('Post Listing')
+    @post = Post.last
+    visit("/#{@post.id}/#{@post.edit_url}")
```
Comment (zspencer) [inline spec/features/edit_listing_spec.rb:11]: Where is your assert?

File: spec/features/edit_listing_spec.rb [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    fill_in('price', with: '190')
+    click_button('Update Listing')
+    expect(page).to have_content("iPad 2")
+    expect(page).to have_content("White 16GB iPad 2. Wifi with Blue cover")
+    expect(page).to have_content("190")
+  end
+
+  after :each do
+    @post.destroy
+  end
```
Comment (zspencer) [inline spec/features/edit_listing_spec.rb:26]: For each model, I'd recommend putting a  `ModelName.destroy_all` in your `config.before` block for your spec_helper.  Reference Thunder for implementation details.

This means you don't have to worry about destroying individual instances of a model on each test run; which decreases the amount of set up and teardown you hve to do to write tests; which (hopefully) results in moar and better tests!

File: spec/features/list_item_spec.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    visit('/')
+  end
+
+  scenario "should update index with link to post" do
+    expect(page).to have_content("iPad")
+  end
+
+  after :each do
+    @post.destroy
+  end
```
Comment (zspencer) [inline spec/features/list_item_spec.rb:20]: Why do you have a before and after when you have a single scenario?

File: spec/features/unique_post_page_spec.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+feature "Unique page for listing" do
+  before :each do
+    visit('/')
+    fill_in('title', with: 'iPad' )
+    fill_in('description', with: 'White 16GB iPad 2. Wifi with Orange cover')
+    fill_in('price', with: '200')
+    click_button('Post Listing')
+    @post = Post.last
+  end
```
Comment (zspencer) [inline spec/features/unique_post_page_spec.rb:11]: I'd really recommend pulling this chunk of code out into a method that you can call in your helpers. Check out thunder's `login_as` function for details.

**Comment by zspencer** [comment]
Looking pretty good. You've got quite a bit of duplication in your test files. I'd extract them to helpers when possible.

**Pull request closed without merging**